sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2914,7 +2914,10 @@ object SQLConf {
     buildConf("spark.sql.defaultColumn.allowedProviders")
       .internal()
       .doc("List of table providers wherein SQL commands are permitted to assign DEFAULT column " +
-        "values. Comma-separated list, whitespace ignored, case-insensitive.")
+        "values. Comma-separated list, whitespace ignored, case-insensitive. If an asterisk " +
+        "appears after any table provider in this list, any command may assign DEFAULT column " +
+        "except ALTER TABLE ... ADD COLUMN. Otherwise, if no asterisk appears, all commands are " +
```

Comment:
nit: let's quote the command with '`'
```
except `ALTER TABLE ... ADD COLUMN`
```

Reply:
Also, it would be great if we can have a simple explanation about why "alter table add column" is special.

Reply:
Good ideas, thanks for the suggestions. Updated the comment.